background.ts

```diff
@@ -324,6 +294,61 @@ Electron.app.whenReady().then(async() => {
   }
 });
 
+/**
+ * Checks for the existence of the 'backend.lock' file and updates the
+ * 'backendIsLocked' variable accordingly. Emits the 'backend-locked-update'
+ * event to notify listeners about the current lock status.
+ *
+ * This function blocks execution until the 'backend.lock' file is no longer
+ * present.
+ */
+function checkForBackendLock() {
+  return new Promise((resolve) => {
+    const intervalId = setInterval(() => {
+      if (!fs.existsSync(path.join(paths.appHome, 'backend.lock'))) {
+        backendIsLocked = '';
+        mainEvents.emit('backend-locked-update', backendIsLocked);
+        clearInterval(intervalId);
+        resolve(backendIsLocked);
+      } else {
+        backendIsLocked = SNAPSHOT_OPERATION;
+        mainEvents.emit('backend-locked-update', backendIsLocked);
+      }
+    }, 1000);
```

Comment:
Does this mean that we add a second to startup even if the lock doesn't exist? (Because we won't run the first iteration until after the first time the interval fires.)